modules/js/src/loader.js

```diff
@@ -0,0 +1,108 @@
+function Loader() {
+    let self = this;
+    let asmPath = "";
+    let wasmPath = "";
+    let simdPath = "";
+    let threadsPath = "";
+    let mtSIMDPath = "";
+    
+    
+    this.judgeWASM = function() {
+        try{
+            let test = WebAssembly;
+            return true;
+        } catch(e) {
+            return false;
+        }
+    }
+
+    this.judgeSIMD = function() {
+        return WebAssembly.validate(new Uint8Array([0,97,115,109,1,0,0,0,1,4,1,96,0,0,3,2,1,0,10,9,1,7,0,65,0,253,15,26,11]));
+    }
+
+    this.judgeThreads = function() {
+        try {
+            let test1 = (new MessageChannel).port1.postMessage(new SharedArrayBuffer(1));
+            let result = WebAssembly.validate(new Uint8Array([0,97,115,109,1,0,0,0,1,4,1,96,0,0,3,2,1,0,5,4,1,3,1,1,10,11,1,9,0,65,0,254,16,2,0,26,11]));
+            return result;
+        } catch(e) {
+            return !1
+        }
+    }
+
+    this.setPaths = function(paths) {
+        if(!(paths instanceof Object)) {
+            throw new Error("The input should be a object that points the path to the OpenCV.js");
+        }
+
+        if ("asm" in paths) {
+            self.asmPath = paths["asm"];
+        }
+
+        if ("wasm" in paths) {
+            self.wasmPath = paths["wasm"];
+        }
+
+        if ("threads" in paths) {
+            self.threadsPath = paths["threads"];
+        }
+
+        if ("simd" in paths) {
+            self.simdPath = paths["simd"];
+        }
+
+        if ("mtSIMD" in paths) {
+            self.mtSIMDPath = paths["mtSIMD"];
+        }
+    }
+
+    this.loadOpenCV = function (onloadCallback) {
```

Comment:
I think you can encapsulate all functions into this method. Do you really need a `loader` object?

Reply:
```js
function loadOpenCV(paths, onloadCallback) {...}
```
It would have simpler usage.

Reply:
The `loader` object is not necessary. I think I can try to encapsulate all functions in `loadOpenCV`.